program/src/main/QuestionSet.vala

```diff
@@ -1,39 +1,67 @@
 using Xml;
 using Gee;
 
-public class QuestionSet : Object {
-    private int question_num; // question number
-    private double total_points; // total points possible
-    private double[] bounds;
-    private int page_num; // which page is this question on?
-    private Gee.HashMap<int, Mark> rubric_pool;
-    private Gee.ArrayList<Question> questions;
-
-    public QuestionSet(int q_num, double total_points, double[] bounds, int page_num, int num_questions) {
-        this.question_num = q_num;
-        this.total_points = total_points;
-        this.bounds = bounds;
-        this.page_num = page_num;
+
+public class QuestionSet : Object{
+    public int question_num; // question number
+    public double total_points; // total points possible
+    public double[] bounds;
+    public int page_num; // which page is this question on?
+    public Gee.HashMap<int, Mark> rubric_pool;
+    public Gee.ArrayList<Question> questions;
+    //private Question currQuestion;
+
+    public QuestionSet(int QNum, double total, double[] b, int pNum){
+        this.question_num = QNum;
+        this.total_points = total;
+        this.bounds = b;
+        this.page_num = pNum;
         this.questions = new Gee.ArrayList<Question>();
-        for(int i = 0; i < num_questions; i++) {
+        this.rubric_pool = new Gee.HashMap<int, Mark>();
+    }
+
+    // Fills question set with new empty questions - HAS to be called before grading starts but can only be called after the number of tests has been determined (aka, end of setup)
+    public void init_question_set(int num) { 
+        for(int i = 0; i < num; i++) { 
             Question q = new Question();
             this.questions.add(q);
         }
-        this.rubric_pool = new Gee.HashMap<int, Mark>();
     }
     
-    public void add_default_marks() {
+    public void addDefaultMarks() {
         this.rubric_pool[0] = new Mark(0, -total_points, "No points", true);
         this.rubric_pool[1] = new Mark(1, -total_points/2, "Half points", true);
         this.rubric_pool[2] = new Mark(2, 0.0, "Full points", true);
     }
+    
+    public Question questionSelect(int num){
+        return questions.get(num);
+    }
 
     public int get_question_num() {
         return this.question_num;
     }
 
+
+    public bool are_bounds_valid() { 
+        if (bounds[0] == 0 && bounds[1] == 0 && bounds[2] == 0 && bounds[3] == 0) { 
+            return true;
+        } else { 
+            return false;
+        } 
+    }
+
+    public void set_points(double points) { 
+        this.total_points = points;
+    }
+
+    public void set_bounds(double[] bounds){
+        this.bounds = bounds;
+    }
+
     public double get_total_points() {
         return this.total_points;
+
```

Comment:
Extra line here 🙂

Reply:
These extra lines will be the death of me lmao they're everywhere. Nice catch - fixed in latest commit 👍